js/components/frame.js

```diff
@@ -337,8 +337,9 @@ class Frame extends ImmutableComponent {
           this.props.frame)
         const key = this.props.frame.get('key')
         const parsedUrl = urlParse(event.url)
+        // don't change url for non-display protocols like mailto
         if (['http:', 'https:', 'about:', 'chrome:', 'chrome-extension:', 'file:',
-             'view-source:'].includes(parsedUrl.protocol)) {
+             'view-source:', 'ftp:', 'data:'].includes(parsedUrl.protocol)) {
```

Comment:
I guess it's a question of the severity of the problem. Changing the url for skype, etc... is a bad user experience, but not changing the protocol for data was a real security issue. Given that, I'd rather blacklist than whitelist or come up with a different solution that doesn't even fire the event for protocols that shouldn't be handled by it